### ISSUE TITLE
[Accessibility] The variation attributes dropdown should also accept aria_label as argument

### ISSUE THREAD

**Issue by Manussakis** [woothemes/woocommerce#60120]
### Prerequisites (mark completed items with an [x]):

- [x] I have checked that my issue type is not listed here https://github.com/woocommerce/woocommerce/issues/new/choose
- [x] My issue is not a security issue, support request, bug report, enhancement or feature request (Please use the link above if it is).

### Issue Description:
The variation attribute dropdown already has a label associated with it. 

<img width="428" height="108" alt="Image" src="https://github.com/user-attachments/assets/7d404e75-d1e0-4c59-b875-de82985bf030" />

This works great for pages that have only one product with variations to be selected, but it can be an issue if the page has multiple products with variations. In that case, multiple fields share the same label, making it difficult to distinguish their functions.

e.g.,

<img width="548" height="854" alt="Image" src="https://github.com/user-attachments/assets/3c0d145e-ba53-4cfc-9d1c-7e98f3fcd853" />

By allowing the dropdown to also receive the `aria-label` value, merchants can add a unique accessible name to them and improve accessibility.

### Component or Page:
Pages: Single Product

### Relevant code snippets:
https://github.com/woocommerce/woocommerce/blob/1894c32b0e91954d510c6a57f16286e6b69a62a6/plugins/woocommerce/includes/wc-template-functions.php#L3345

### Recommended fix:
The `args` parameter of the `wc_dropdown_variation_attribute_options` function should also accept a property called `aria_label`. The value should be used to populate the `aria-label` HTML attribute of the [`<select>`](https://github.com/woocommerce/woocommerce/blob/1894c32b0e91954d510c6a57f16286e6b69a62a6/plugins/woocommerce/includes/wc-template-functions.php#L3384) element.

**Comment by PanosSynetos** [comment]
I've assigned this to @Manussakis - we've discussed for this assignment.
